packages/fetchai/connections/http/connection.py

```diff
@@ -0,0 +1,384 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""HTTP connection, channel, server, and handler"""
+
+import asyncio
+import json
+import logging
+import threading
+from asyncio import CancelledError
+from http.server import BaseHTTPRequestHandler, HTTPServer
+from typing import Optional, Set, cast
+from urllib.parse import parse_qs, urlparse
+from uuid import uuid4
+
+import yaml
+
+from aea.configurations.base import ConnectionConfig, PublicId
+from aea.connections.base import Connection
+from aea.mail.base import Address, Envelope, EnvelopeContext, URI
+
+logger = logging.getLogger(__name__)
+
+
+class HTTPChannel:
+    """A wrapper for an RESTful API with an internal HTTPServer."""
+
+    def __init__(
+        self,
+        address: Address,
+        host: str,
+        port: int,
+        api_spec: str,
+        connection_id: PublicId,
+        excluded_protocols: Optional[Set[PublicId]] = None,
```

Comment:
you don't need the excluded protocols here. It is just needed on the `connection` object

Reply:
I have removed from HTTPChannel, in next commit.